tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py

```diff
@@ -126,47 +139,56 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
-        all_pod_obj = pod.get_all_pods(
-            namespace=defaults.ROOK_CLUSTER_NAMESPACE
-        )
-        for pod_obj in all_pod_obj:
-            if '-1-deploy' and 'ocs-deviceset' not in pod_obj.name:
-                try:
-                    helpers.wait_for_resource_state(
-                        resource=pod_obj, state=constants.STATUS_RUNNING,
-                        timeout=60
-                    )
-                except ResourceWrongStatusException:
-                    # 'rook-ceph-crashcollector' on the failed node stucks at
-                    # pending state. BZ 1810014 tracks it.
-                    # Ignoring 'rook-ceph-crashcollector' pod health check as
-                    # WA and deleting its deployment so that the pod
-                    # disappears. Will revert this WA once the BZ is fixed
-                    if 'rook-ceph-crashcollector' in pod_obj.name:
-                        ocp_obj = ocp.OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-                        pod_name = pod_obj.name
-                        deployment_name = '-'.join(pod_name.split("-")[:-2])
-                        command = f"delete deployment {deployment_name}"
-                        ocp_obj.exec_oc_cmd(command=command)
-                        log.info(f"Deleted deployment for pod {pod_obj.name}")
+            # Check all OCS pods status, they should be in running state
+            all_pod_obj = pod.get_all_pods(
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE
+            )
+            for pod_obj in all_pod_obj:
+                if '-1-deploy' and 'ocs-deviceset' not in pod_obj.name:
+                    try:
+                        helpers.wait_for_resource_state(
+                            resource=pod_obj, state=constants.STATUS_RUNNING,
+                            timeout=60
+                        )
+                    except ResourceWrongStatusException as ex:
+                        # 'rook-ceph-crashcollector' on the failed node stucks at
+                        # pending state. BZ 1810014 tracks it.
+                        # Ignoring 'rook-ceph-crashcollector' pod health check as
+                        # WA and deleting its deployment so that the pod
+                        # disappears. Will revert this WA once the BZ is fixed
+                        if 'rook-ceph-crashcollector' in pod_obj.name:
+                            ocp_obj = ocp.OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+                            pod_name = pod_obj.name
+                            deployment_name = '-'.join(pod_name.split("-")[:-2])
+                            command = f"delete deployment {deployment_name}"
+                            ocp_obj.exec_oc_cmd(command=command)
+                            log.info(f"Deleted deployment for pod {pod_obj.name}")
+                        else:
+                            raise
+        except ResourceWrongStatusException:
+            if failure == "shutdown":
```

Comment:
Hi @PrasadDesala , please review if this change is ok with this test case you created

Reply:
LGTM